frontend/src/metabase/admin/settings/components/SettingsXrayForm.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import SettingsSetting from 'metabase/admin/settings/components/SettingsSetting'
+import cx from 'classnames'
+
+import Icon from 'metabase/components/Icon'
+
+import COSTS from 'metabase/xray/costs'
+
+const SettingsXrayForm = ({ settings, elements, updateSetting }) => {
+    let maxCost = Object.assign({}, ...elements.filter(e => e.key === 'xray-max-cost',))
```

Comment:
An odd pattern because it encourages mutation (see line 15 where `maxCost.value = 'extended'`). I would rather do simply something like
```
const maxConst = elements.find(criterion) || defaultValue
```